src/main/proto/eventtemplates/common_template.proto

```diff
@@ -32,18 +32,21 @@ message CommonTemplate {
     option (.wfa.measurement.api.v2alpha.event_field) = {
       display_name: "Viewability"
     };
-
-    // Whether viewability was able to be measured in the event.
-    uint32 measurable = 1;
-
-    // More than 50% of the ad was displayed in the browser for 1 second for
-    // display ads or 2 seconds for video ads. Aka the MRC viewable
-    // definition.
-    uint32 viewable_50_percent_plus = 2;
-
-    // 100% of the ad was displayed in the browser for at least 1 second for
-    // display ads or 2 seconds for video ads.
-    uint32 viewable_100_percent = 3;
+    
+    enum Value {
+      // More than 0% of the ad was displayed in the browser for at least 1
+      // second for display ads or at least 2 seconds for video ads.
+      VIEWABLE_0_PERCENT_PLUS = 1;
```

Comment:
Since this is now an enum, I think these should be 

VIEWABLE_0_PERCENT_TO_50_PERCENT = 1;
VIEWABLE_50_PERCENT_TO_100_PERCENT = 2;
VIEWABLE_100_PERCENT = 3;

Since only one of the values will be set.

Reply:
@uakyol - your suggestion doesn't match our use case. The enumeration options for viewability are intentionally overlapping. Viewability is a measurement standard which encompasses two elements:
1. % of the ad visible on-screen to the end user
2. amount of time that the ad was on-screen
 
When we select common_template.viewability == "0_PER_CENT_PLUS" in the filter then the expectation is that all impressions where greater than 0 pixels of the ad were viewable for greater than zero seconds should be returned. (i.e. not just impressions where between 0 and 50% of the ad was visible).


VIEWABLE_100_PER_CENT -- Return impressions where 100% of the ad was visible for at least 1 second for display ads or 2 seconds for video ads. AKA the new MRC viewability standard.

VIEWABLE_50_PER_CENT_PLUS -- Return impressions where greater than 50% of the ad was visible for at least 1 second for display ads and 2 seconds for video ads. AKA the old MRC viewabilty standard. 

0_PER_CENT_PLUS -- Return impressions where greater than 0% of the ad was visible for greater than 0 seconds. Note I've removed the prefix "VIEWABLE" from this option to indicate that this option is not an MRC Viewable definition and has a different time element.

It would also be possible to set no filter which should return all impressions regardless of viewability; and it would be possible to return "MRC Non-Viewable" events i.e. those that don't meet the MRC standard using a != operand. i.e. common_template.viewability != "VIEWABLE_100_PER_CENT".

Reply:
@khisba - Lets talk more about the 2 dimensions and what should they exactly mean in the Thursday's call. But the issue with enums being _PLUS or _BETWEEN is not actually contradictory. 

These event templates are a schema for an event. So for this enum to have overlapping fields means for one event more than one of these enums have to be set at the same time. Which is not possible since they are enums.

If we change the field names to "ranges", we will achieve the same effect as you outlined above with these filters: 

**Previously VIEWABLE_100_PER_CENT** : 
eventFilter => common_template.viewability.value == VIEWABLE_100_PERCENT

**Previously VIEWABLE_50_PER_CENT_PLUS** : 
eventFilter => common_template.viewability.value == VIEWABLE_100_PERCENT || common_template.viewability.value == VIEWABLE_50_PERCENT_TO_100_PERCENT


**Previously 0_PER_CENT_PLUS** : 
eventFilter => common_template.viewability.value == VIEWABLE_100_PERCENT || common_template.viewability.value == VIEWABLE_50_PERCENT_TO_100_PERCENT || VIEWABLE_0_PERCENT_TO_50_PERCENT


**No eventFilter set** => Returns everything